internal/alloydbutil/engine.go

```diff
@@ -85,16 +87,20 @@ func (p *PostgresEngine) Close() {
 // getUser retrieves the username, a flag indicating if IAM authentication
 // will be used and an error.
 func getUser(ctx context.Context, config engineConfig) (string, bool, error) {
-	// If neither user nor password are provided, retrieve IAM email.
-	if config.user == "" && config.password == "" {
+	if config.user != "" && config.password != "" {
```

Comment:
I'm a little confused why this section is changing to support a Chat Message History? If it's unrelated, probably best as a separate PR